### PR TITLE
rename to libvfio-user

### LINKED ISSUE #121: rename muser?
Should we rename the project, library and API? The current name is no longer well-aligned with the current implementation.

### PR DESCRIPTION
**Pull request by jlevon**
The `muser` name no longer reflects the implementation, and will just serve to
confuse. Bite the bullet now, and rename ourselves to reflect the actual
implementation.

Fixes #121

Signed-off-by: John Levon <john.levon@nutanix.com>

### CODE REVIEW

**Review by tmakatos — COMMENTED** [review]
No sure about the `vu` prefix, if `vfio_user` far too long? What about something in between, e.g. `vfusr`? Also, we should rename the project on github, can we redirect nutanix/muser to nutanix/libvfio_user?

**Comment by jlevon** [comment]
IIRC github does redirects (I hope so anyway - if not we could have a stub redirecting repo).

Now, down to the serious bike-shedding :)

vfio_user is a) too long and b) is in the protocol namespace. The latter in particular could confusing fast.

So I'd prefer something visually distinct from the protocol header. Time for a slack poll?

We could have a "lib" prefix: libvu_, libvfu_, libvfusr_, libvfuser_...
Or no prefix: vu_, lvu_, vfusr, vfuser_, ...

**Comment by tmakatos** [comment]
> IIRC github does redirects (I hope so anyway - if not we could have a stub redirecting repo).
> 
> Now, down to the serious bike-shedding :)
> 
> vfio_user is a) too long and b) is in the protocol namespace. The latter in particular could confusing fast.
> 
> So I'd prefer something visually distinct from the protocol header. Time for a slack poll?
> 
> We could have a "lib" prefix: libvu_, libvfu_, libvfusr_, libvfuser_...
> Or no prefix: vu_, lvu_, vfusr, vfuser_, ...

I like something similar to `vfusr` or `vfuser`, but since this is clearly a matter of personal preference I believe a Slack poll would be the best way to decide :-).

**Comment by jlevon** [comment]
Here's the vfu_ version!

File: README.md [hunk 1]
```diff
@@ -1,80 +1,98 @@
...
+over a UNIX socket to a server. This library, `libvfio-user`, provides an API
+for implementing such servers.
 
-MUSER is a framework that allows implementing PCI devices under the [vfio-user
-protocol](https://lists.gnu.org/archive/html/qemu-devel/2020-11/msg02458.html).
-MUSER is the _backend_ part of the vfio-user protocol, the frontend part is
-implemented by Oracle in https://github.com/oracle/qemu/tree/vfio-user-v0.1.
+[VFIO](https://www.kernel.org/doc/Documentation/vfio.txt) is a kernel facility
+for providing secure access to PCI devices in userspace (including pass-through
+to a VM). With `vfio-user`, instead of talking to the kernel, all operations are
```
Comment (swapnili) [inline README.md:12]: I think 'all operations' is incorrect. vfio-user do not support iommu groups or any iommu related functionalities which are exposed by vfio over ioctls.

File: README.md [hunk 2]
```diff
@@ -1,80 +1,98 @@
...
+interrupts simply by writing to it. This can be much more expensive compared to
+triggering interrupts from the kernel, however this performance penalty is
+perfectly acceptable when prototyping the functional aspect of a device driver.
 
 
 Building muser
 ==============
 
+Build requirements are `cmake` (v2 or above) and `libjson-c-dev` /
+`libjson-c-devel`.
```
Comment (swapnili) [inline README.md:67]: Should we also add libcmocka-dev to requirements list? I remember my local build failed and was expecting libcmocka-dev installed.

**Review by tmakatos — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (tmakatos) [reply README.md:67]: Yes, we should.

**Review by swapnili — APPROVED** [review]
Looks good to me. Apart from some minor comments.
Thanks

File: README.md (hunk 1 above)
Comment (jlevon) [reply README.md]: That's not really what I meant; I added some clarifying comments.

File: README.md (hunk 2 above)
Comment (jlevon) [reply README.md]: done

**Review by swapnili — APPROVED** [review]

**Pull request merged**